src/pages/Designer.js

```diff
@@ -905,11 +911,17 @@ const Designer = ({ update, viewMode = false, sharedMetadata = undefined }) => {
                     }
                     setIsGatewayNodeEnabled(true);
                 } else if (key.toLowerCase().includes('database')) {
-                    databaseId++;
+                    const id = key.split('_');
+                    const numberId = parseInt(id[1], 10);
+                    max_databaseId = Math.max(numberId, max_databaseId);
                 } else if (key.toLowerCase().includes('group')) {
-                    groupId++;
+                    const id = key.split('_');
+                    const numberId = parseInt(id[1], 10);
+                    max_groupId = Math.max(numberId, max_groupId);
                 } else if (key.toLowerCase().includes('dummy')) {
-                    dummyId++;
+                    const id = key.split('_');
+                    const numberId = parseInt(id[1], 10);
```

Comment:
default radix for parseInt is 10 right, no need to specify explicitly in the second argument?
please update all 3 usages